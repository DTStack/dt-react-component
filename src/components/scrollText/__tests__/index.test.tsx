import React from 'react';
import ScrollText from '../index';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('test ScrollText suite', () => {
    let dom, element;
    beforeEach(() => {
        dom = render(<ScrollText value="scroll" className="testScrollText" />);
        element = dom.getByTestId('test-scroll-text');
    });
    afterEach(() => {
        cleanup();
    });
    test('snapshot match', () => {
        expect(element).toMatchSnapshot();
    });
    test('should render the correct text', () => {
        expect(element).toBeInTheDocument();
        expect(element).toHaveValue('scroll');
    });
    test('should render ScrollText custom className', () => {
        expect(element).toHaveClass('testScrollText');
    });
});
