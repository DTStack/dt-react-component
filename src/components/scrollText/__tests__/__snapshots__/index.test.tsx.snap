// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ScrollText suite snapshot match 1`] = `
<input
  class="cell-input testScrollText"
  data-testid="test-scroll-text"
  readonly=""
  style="height: 28px; margin: 5px 5px 5px 0px; width: 100%; text-align: left; background-color: transparent; background-image: none;"
  title="scroll"
  value="scroll"
/>
`;
